src/libmongoc/src/mongoc/mongoc-ts-pool.c

```diff
@@ -0,0 +1,117 @@
+#include "./mongoc-ts-pool-private.h"
+#include "common-thread-private.h"
+
+#include "bson/bson.h"
+
+typedef struct pool_node {
+   struct pool_node *next;
+   max_align_t data[1];
```

Comment:
`max_align_t` ensures that that data member is aligned to the maximum possible data alignment of the target CPU architecture, such that when I over-allocate the struct in `alloc_item` and then type-pun that data member to some `T`, it is guaranteed to be at-least as aligned as `T`. If I just made it `char` or something arbitrary, it could be under-aligned and then type-punning the data member would be UB. If there's another way to detect the max-alignment type, that would work in place of the `max_align_t`.